ert/serialization/_serializer.py

```diff
@@ -70,3 +71,26 @@ async def decode_from_path(
         async with aiofiles.open(path, mode="rt", encoding="utf-8") as filehandle:
             contents = await filehandle.read()
             return self.decode(contents, *args, **kwargs)
+
+
+class _ecl_sum_serializer(Serializer):
+    def encode(self, obj: Any, *args: Any, **kwargs: Any) -> str:
+        raise NotImplementedError("not implemented")
+
+    def decode(self, series: str, *args: Any, **kwargs: Any) -> Any:
+        raise NotImplementedError("not implemented")
+
+    async def encode_to_path(
+        self, obj: Any, path: Union[str, Path], *args: Any, **kwargs: Any
+    ) -> None:
+        raise NotImplementedError("not implemented")
+
+    async def decode_from_path(
+        self, path: Union[str, Path], *args: Any, **kwargs: Any
+    ) -> Any:
+        key = kwargs.get("key", None)
+        if key is None:
+            raise ValueError("key must be provided as a keyword argument")
+
+        eclsum = EclSum(str(path))
+        return dict(zip(map(str, eclsum.dates), eclsum.numpy_vector(key)))
```

Comment:
Does `eclsum` always need to come with dates as I recall having some index based data too, but might be wrong @berland ?

Reply:
I have never heard of `EclSum` without dates, but will not boldly claim that it cannot happen.